packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -789,18 +790,15 @@ export class TeamsBotImpl {
       );
       botAuthCreds = await AADRegistration.registerAADAppAndGetSecretByAppStudio(
         appStudioToken!,
-        aadDisplayName
+        aadDisplayName,
+        this.config.scaffold.objectId
       );
 
-      if (!this.config.scaffold.botId) {
-        this.config.scaffold.botId = botAuthCreds.clientId;
-      }
-      if (!this.config.scaffold.botPassword) {
-        this.config.scaffold.botPassword = botAuthCreds.clientSecret;
-      }
-      if (!this.config.scaffold.objectId) {
-        this.config.scaffold.objectId = botAuthCreds.objectId;
-      }
+      this.config.scaffold.botId = botAuthCreds.clientId ?? this.config.scaffold.botId;
```

Comment:
Why change it like this? The logic is different.

Reply:
Can you explain more about the general logic here first so that we can see if this change is reasonable?

Reply:
@IvanJobs for the above has the logic `if (!this.config.scaffold.botAADCreated()) {` , and no need to judge again, and the bothAuthCreds's value should override the value in config.scaffold

Reply:
@eriolchan explained in below comments